### PR TITLE
Get HW info from DMI/SMBIOS

### LINKED ISSUE #124: use `ghw` to get `board` and `product` inventory info for the agent
see example : https://github.com/jaypipes/ghw#baseboard

```
package main

import (
	"fmt"

	"github.com/jaypipes/ghw"
)

func main() {
	baseboard, err := ghw.Baseboard()
	if err != nil {
		fmt.Printf("Error getting baseboard info: %v", err)
	}

	fmt.Printf("%v\n", baseboard)
}
```

https://github.com/jaypipes/ghw/blob/main/pkg/baseboard/baseboard_linux.go
https://github.com/jaypipes/ghw/blob/main/pkg/product/product_linux.go

### PR DESCRIPTION
**Pull request by glimchb**
Fixes #124

Signed-off-by: Boris Glimcher <Boris.Glimcher@emc.com>

### CODE REVIEW

File: sztp-agent/pkg/secureagent/utils.go [hunk 1]
```diff
@@ -94,8 +95,10 @@ func (a *Agent) doTLSRequest(input string, url string, empty bool) (*BootstrapSe
 func generateInputJSONContent() string {
 	osName := replaceQuotes(strings.Split(linesInFileContains(OS_RELEASE_FILE, "NAME"), "=")[1])
 	osVersion := replaceQuotes(strings.Split(linesInFileContains(OS_RELEASE_FILE, "VERSION"), "=")[1])
-
-	// dmidecode.Dmidecode(true))  // This is one possibility to get hw information
+	baseboard, err := ghw.Baseboard()
+	if err != nil {
+		fmt.Printf("Error getting baseboard info: %v", err)
```
Comment (glimchb) [inline sztp-agent/pkg/secureagent/utils.go:100]: @alknopfler should we return an error here ?

File: sztp-agent/pkg/secureagent/utils.go [hunk 2]
```diff
@@ -94,8 +95,10 @@ func (a *Agent) doTLSRequest(input string, url string, empty bool) (*BootstrapSe
 func generateInputJSONContent() string {
 	osName := replaceQuotes(strings.Split(linesInFileContains(OS_RELEASE_FILE, "NAME"), "=")[1])
 	osVersion := replaceQuotes(strings.Split(linesInFileContains(OS_RELEASE_FILE, "VERSION"), "=")[1])
-
-	// dmidecode.Dmidecode(true))  // This is one possibility to get hw information
+	baseboard, err := ghw.Baseboard()
+	if err != nil {
+		log.Printf("Error getting baseboard info: %v", err)
```
Comment (glimchb) [inline sztp-agent/pkg/secureagent/utils.go:100]: @alknopfler should we return an error here instead of just logging it ?

Comment (alknopfler) [reply sztp-agent/pkg/secureagent/utils.go:100]: makes sense return 2 values (value, and error), but then, in the invocation (agent.go) we should do it out of the struct to control it, and returning in newAgent the error too to be handle by the CLI in cmd package. Something like:

```
func NewAgent(bootstrapURL, serialNumber, devicePassword, devicePrivateKey, deviceEndEntityCert, bootstrapTrustAnchorCert string) (*Agent, error) {
	inputContent,err := generateInputJSONContent()
	if err!= nil{ 
	    return nil, err
	}    
	return &Agent{
		BootstrapURL:                  bootstrapURL,
		SerialNumber:                  serialNumber,
		DevicePassword:                devicePassword,
		DevicePrivateKey:              devicePrivateKey,
		DeviceEndEntityCert:           deviceEndEntityCert,
		BootstrapTrustAnchorCert:      bootstrapTrustAnchorCert,
		ContentTypeReq:                CONTENT_TYPE_YANG,
		InputJSONContent:              inputContent,
		DhcpLeaseFile:                 DHCLIENT_LEASE_FILE,
		ProgressJSON:                  ProgressJSON{},
		BootstrapServerRedirectInfo:   BootstrapServerRedirectInfo{},
		BootstrapServerOnboardingInfo: BootstrapServerOnboardingInfo{},
	}, nil
}
```
and in the cmd package with the cli the agent instantiation should handle the error creating the agent :

```
cmd := &cobra.Command{
		Use:   "daemon",
		Short: "Run the daemon command",
		RunE: func(cmd *cobra.Command, args []string) error {
			a, err := secureagent.NewAgent(bootstrapURL, serialNumber, devicePassword, devicePrivateKey, deviceEndEntityCert, bootstrapTrustAnchorCert)
                         if err != nil {
                             return err
                         }
			return a.RunCommandDaemon()
		},
	}
```

**Review by alknopfler — CHANGES_REQUESTED** [review]

File: sztp-agent/pkg/secureagent/utils.go (hunk 2 above)
Comment (glimchb) [reply sztp-agent/pkg/secureagent/utils.go:101]: since `HwModel` is optional, will just use empty string `""` in case of an error

**Review by alknopfler — APPROVED** [review]
LGTM

**Pull request merged**